packages/edit-site/src/components/page-patterns/index.js

```diff
@@ -187,11 +197,13 @@ function Author( { item, viewType } ) {
 
 function Title( { item } ) {
 	const isUserPattern = item.type === PATTERN_TYPES.user;
+	const isTemplatePart = item.type === TEMPLATE_PART_POST_TYPE;
 	const { onClick } = useLink( {
 		postType: item.type,
-		postId: isUserPattern ? item.id : item.name,
+		postId: isUserPattern || isTemplatePart ? item.id : item.name,
```

Comment:
I've looked into this and this is what I see:

|                    | id                                     | name                                           | title                                             |
| ------------------ | -------------------------------------- | ---------------------------------------------- | ------------------------------------------------- |
| Template part      | `twentytwentyfour//footer`             | `twentytwentyfour//footer`                     | Footer                                            |
| User template part | `twentytwentyfour//user-template-part` | `twentytwentyfour//user-template-part`         | user template part                                |
| Pattern            | -                                      | `twentytwentyfour/text-title-left-image-right` | Title text and button on left with image on right |
| User pattern       | 1836                                   | `my-pattern`                                   | My pattern                                        |

So, it seems that:

- `id` or `name` can be used to provide links to template parts (trying to understand if this is a bug in current `trunk`, but it isn't).
- This link is only provided for items that are not patterns by the theme: it could be simplified to always return `item.id`
- The proposed change is also fine as it is.

Reply:
That's not entirely correct. Template parts don't have "name". You can try this locally to check `wp.data.select('core').getEntityRecords( 'postType', 'wp_template_part' );` 

The lack of fallback to id was also what was triggering the React key warning raised by Ella.

Reply:
Not sure what other processing we do and how the data differs in other parts of the codebase (still reviewing parts of the PR and catching up), but the `item` processed by DataViews does have that data (compiled the table above using a console.log within the `Title` component to).

I've also tested that links from the title worked fine in `trunk` for template parts — if they didn't have the `name`, it would have failed and would be something to cherry-pick for 6.6 I presume.

Reply:
Yes, that's actually the purpose of this PR. We should be using the default objects and not create local formats.

Reply:
I only have `packages/editor/src/components/post-actions/actions.js` for review. I need to be somewhere else now, will review that one later. Changes are looking good so far.

Reply:
done the review